### PR TITLE
fix: Allow dynamic value change for range slider #1154

### PR DESCRIPTION
**Pull request by marek-mihok**
**The PR fulfills these requirements:** (check all the apply)

- [x] It's submitted to the `main` branch.
- [x] When resolving a specific issue, it's referenced in the PR's title (e.g. `feat: Add a button #xxx`, where "xxx" is the issue number).
- [x] When resolving a specific issue, the PR description includes `Closes #xxx`, where "xxx" is the issue number.
- [x] If changes were made to `ui` folder, unit tests (`make test`) still pass.
- [x] New/updated tests are included

___

This PR partially resolves issue #1154.

NOTE: Do not forget to update status in [this comment](https://github.com/h2oai/wave/issues/1154#issuecomment-1252010473) once merged.

### CODE REVIEW

File: ui/src/range_slider.tsx [hunk 1]
```diff
@@ -54,17 +54,26 @@ export interface RangeSlider {
...
+    [valRange, setValRange] = React.useState<[U, U]>([min_value, max_value]),
+    onChange = React.useCallback((_val: U, val_range?: [U, U]) => {
+      if (val_range) {
+        wave.args[name] = val_range
+        model.min_value = val_range[0]
+        model.max_value = val_range[1]
+        setValRange(val_range)
+      }
+      // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [name]),
```
Comment (mturoci) [inline ui/src/range_slider.tsx:67]: Any reason for not including `model` in dep array?

File: ui/src/range_slider.tsx [hunk 2]
```diff
@@ -54,17 +54,26 @@ export interface RangeSlider {
...
     onChanged = React.useCallback(() => { if (trigger) wave.push() }, [trigger])
 
   React.useEffect(() => {
-    wave.args[name] = [
-      typeof min_value == 'number' && min_value > min && min_value <= max ? min_value : min,
-      typeof max_value == 'number' && max_value > min && max_value <= max ? max_value : max,
-    ]
+    const lowerValue = typeof min_value == 'number' && min_value > min && min_value <= max ? min_value : min
+    const value = typeof max_value == 'number' && max_value > min && max_value <= max ? max_value : max
```
Comment (mturoci) [inline ui/src/range_slider.tsx:72]: Nit: naming

```suggestion
    const upperValue = typeof max_value == 'number' && max_value > min && max_value <= max ? max_value : max
```

File: ui/src/range_slider.tsx [hunk 3]
```diff
@@ -54,17 +54,26 @@ export interface RangeSlider {
...
+    wave.args[name] = [lowerValue, value]
+    setValRange([lowerValue, value])
     // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [])
+  }, [min_value, max_value])
```
Comment (mturoci) [inline ui/src/range_slider.tsx:76]: We probably want to include full dep array here, right? Same as `spinbox` PR.

**Review by mturoci — CHANGES_REQUESTED** [review]

File: ui/src/range_slider.tsx (hunk 2 above)
Comment (marek-mihok) [reply ui/src/range_slider.tsx:72]: I've tried to be consistent with FluentUI's props `lowerValue` and `value`, however `lowerValue` and `upperValue` makes much more sense.

**Review by mturoci — APPROVED** [review]
LGTM

**Pull request merged**